File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    findMin(nums) {
        let left = 0;
        let right = nums.length - 1;
        let middle;
        let res = nums[0];


        while (left <= right){
            if (nums[left] <= nums[right]){
                res = Math.min(res, nums[left]);
                break;
            }

            middle = left + Math.floor((right - left) / 2);
            res = Math.min(res, nums[middle]);

            if (nums[middle] >= nums[left]){
                left = middle + 1;
            }
            else {
                right = middle - 1;
            }
        }

        return res;
    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.cs
public class Solution {
    public int[] ProductExceptSelf(int[] nums) {
        int zeroCount = 0;
        int product = 1;

        for (int i = 0; i < nums.Length; i++){
            if (nums[i] == 0)   zeroCount++;
            else    product *= nums[i];
        }

        if (zeroCount >= 2) return new int[nums.Length];

        int[] res = new int[nums.Length];
        
        for (int i = 0; i < nums.Length; i++){
            if (zeroCount == 1){
                res[i] = nums[i] == 0 ? product : 0;
            }
            else {
                res[i] = product / nums[i];
            }
        }

        return res;
    }
}
